core-processor/src/ext.rs

```diff
@@ -962,18 +1143,26 @@ impl Externalities for Ext {
     }
 
     fn reply_push_input(&mut self, offset: u32, len: u32) -> Result<(), Self::FallibleError> {
-        let range = self.context.message_context.check_input_range(offset, len);
-        self.charge_gas_if_enough(
-            self.context
-                .costs
-                .syscalls
-                .gr_reply_push_input_per_byte
-                .cost_for(range.len().into()),
-        )?;
-
-        self.context.message_context.reply_push_input(range)?;
-
-        Ok(())
+        self.with_changes(|mutator| {
+            let range = mutator
```

Comment:
just for consistency: either move it out of `with_changes` or insert the same `CheckedRange` instantiation call into `with_changes` for `send_push_input`